scripts/scraper-ng/vfile-reporter-summary.js

```diff
@@ -0,0 +1,133 @@
+const chalk = require("chalk");
+const statistics = require("vfile-statistics");
+
+/**
+ * Generate a summary report of VFile messages.
+ *
+ * @param {(VFile|Array.<VFile>|Error)} files - `VFile`, `Array.<VFile>`, or `Error`
+ * @param {Object} options
+ * @returns A string
+ */
+function reporter(files) {
+  // Undefined or `null`
+  if (!files) {
+    return "";
+  }
+
+  // A single Error
+  if ("name" in files && "message" in files) {
+    // We might want to implement the full reporter API at some point, but not now.
+    throw Error("Not implemented");
+  }
+
+  // A single file
+  if (!("length" in files)) {
+    files = [files];
+  }
+
+  // Actually generate the summary
+  return summarize(files);
+}
+
+function summarize(files) {
+  const count = countRules(files);
+
+  const sortedRules = Object.entries(count)
+    .map(([ruleId, details]) => ({
+      ruleId,
+      ...details
+    }))
+    .sort((a, b) => a.count - b.count)
+    .reverse();
```

Comment:
For the sort and reverse, I can reverse sort directly and I'll apply the change. I just thought it was a little more explicit that I was doing a descending rather than ascending sort by doing it the way I did.

For your other comment, I'm not sure what you mean. Could you show me an example of what you mean by `countRules` constructing an array?

Reply:
>  I just thought it was a little more explicit that I was doing a descending rather than ascending sort by doing it the way I did.

Yes, maybe you're right.

> For your other comment, I'm not sure what you mean. Could you show me an example of what you mean by countRules constructing an array?

Having thought about this a bit more, your way is probably best. So forget I said this :).